src/lqcut.jl

```diff
@@ -1,54 +1,46 @@
+    lq_norm::Vector{<:AbstractFloat}, dt_eff::Vector{<:Unitful.RealOrRealQuantity}, e_cal::Vector{<:Unitful.Energy{<:Real}}, DEP_µ::Unitful.AbstractQuantity, DEP_σ::Unitful.AbstractQuantity; 
+    ctc_dep_edgesigma::Float64=3.0 , ctc_driftime_cutoff_method::Symbol=:percentile, lq_outlier_sigma::Float64 = 2.0, drift_time_outlier_sigma::Float64 = 2.0, prehist_sigma::Float64=2.5, lq_e_corr_expression::Union{String,Symbol}="(lq / e)", dt_eff_expression::Union{String,Symbol}="(qdrift / e)" ,ctc_dt_eff_low_quantile::Float64=0.15, ctc_dt_eff_high_quantile::Float64=0.95, pol_fit_order::Int=1) 
 
 Perform the drift time correction on the LQ data using the DEP peak. The function cuts outliers in lq and drift time, then performs a linear fit on the remaining data. The data is Corrected by subtracting the linear fit from the lq data.
 # Returns
     * `result`: NamedTuple of the function used for lq classifier construction
     * `report`: NamedTuple of the histograms used for the fit, the cutoff values and the DEP edges
 """
-function lq_drift_time_correction(
+function lq_ctc_correction(
     lq_norm::Vector{<:AbstractFloat}, dt_eff::Vector{<:Unitful.RealOrRealQuantity}, e_cal::Vector{<:Unitful.Energy{<:Real}}, DEP_µ::Unitful.AbstractQuantity, DEP_σ::Unitful.AbstractQuantity; 
-    DEP_edgesigma::Float64=3.0 , mode::Symbol=:percentile, drift_cutoff_sigma::Float64 = 2.0, prehist_sigma::Float64=2.5, e_expression::Union{String,Symbol}="e", dt_eff_low_quantile::Float64=0.15, dt_eff_high_quantile::Float64=0.95)
-
-    # get energy units to remve later
-    e_unit = unit(first(e_cal))
+    ctc_dep_edgesigma::Float64=3.0 , ctc_driftime_cutoff_method::Symbol=:percentile, lq_outlier_sigma::Float64 = 2.0, drift_time_outlier_sigma::Float64 = 2.0, prehist_sigma::Float64=2.5, lq_e_corr_expression::Union{String,Symbol}="(lq / e)", dt_eff_expression::Union{String,Symbol}="(qdrift / e)" ,ctc_dt_eff_low_quantile::Float64=0.15, ctc_dt_eff_high_quantile::Float64=0.95, pol_fit_order::Int=1) 
 
     # calculate DEP edges
-    DEP_left = DEP_µ - DEP_edgesigma * DEP_σ
-    DEP_right = DEP_µ + DEP_edgesigma * DEP_σ
+    DEP_left = DEP_µ - ctc_dep_edgesigma * DEP_σ
+    DEP_right = DEP_µ + ctc_dep_edgesigma * DEP_σ
 
     # cut data to DEP peak
     lq_DEP = lq_norm[DEP_left .< e_cal .< DEP_right]
     dt_eff_DEP = ustrip.(dt_eff[DEP_left .< e_cal .< DEP_right])
 
-    # Calculate range to truncate lq data for outlier removal
-    ideal_bin_width = get_friedman_diaconis_bin_width(lq_DEP)
-    lq_prehist = fit(Histogram, lq_DEP, range(minimum(lq_DEP), maximum(lq_DEP), step=ideal_bin_width))
-    lq_prestats = estimate_single_peak_stats(lq_prehist)
-    lq_start = lq_prestats.peak_pos - prehist_sigma * lq_prestats.peak_sigma
-    lq_stop = lq_prestats.peak_pos + prehist_sigma * lq_prestats.peak_sigma
+    lq_precut = cut_single_peak(lq_DEP, minimum(lq_DEP), maximum(lq_DEP))
```

Comment:
The `cut_single_peak` function offers a couple `kwargs` which can be tuned. You might want to have a look into that if you want to parse them through here. 
Also, maybe you can find preset parameters for the big windowing rather than just `minimum(lq_DEP)` and `maximum(lq_DEP)`